test/rsspp_parser.cpp

```diff
@@ -378,3 +385,120 @@ TEST_CASE("Feed authors and source authors in atom feed", "[rsspp::Parser]")
 	REQUIRE(f.items[8].description == "Both entry authors should be used.");
 	REQUIRE(f.items[8].author == "Entry Author 1, Entry Author 2");
 }
+
+TEST_CASE("parse_url() extracts etag and lastmodified data", "[rsspp::Parser]")
+{
+	using namespace newsboat;
+
+	auto feed_xml = test_helpers::read_binary_file("data/atom10_1.xml");
+
+	auto& test_server = test_helpers::HttpTestServer::get_instance();
+	auto mock_registration = test_server.add_endpoint("/feed", {}, 200, {
+		{"content-type", "text/xml"},
+		{"ETag", "returned-etag"},
+		{"Last-Modified", "Wed, 21 Oct 2015 07:28:00 GMT"},
+	}, feed_xml);
+	const auto address = test_server.get_address();
+	const auto url = strprintf::fmt("http://%s/feed", address);
+
+	rsspp::Parser parser;
+	CurlHandle easyhandle;
+	parser.parse_url(url, easyhandle);
+
+	REQUIRE(parser.get_etag() == "returned-etag");
+	REQUIRE(parser.get_last_modified() == 1445412480);
+}
+
+// Placeholders:
+// %s: encoding
+// %s: feed title
+constexpr auto atom_feed_with_encoding =
+	R"(<?xml version="1.0" encoding="%s"?>)"
+	R"(<feed xmlns="http://www.w3.org/2005/Atom" xml:lang="en">)"
+	R"(<id>tag:example.com</id>)"
+	R"(<title type="text">%s</title>)"
+	R"(<updated>2008-12-30T18:26:15Z</updated>)"
+	R"(<entry>)"
+	R"(<id>tag:example.com,2008-12-30:/atom_testing</id>)"
+	R"(<title>regular title</title>)"
+	R"(<updated>2008-12-30T20:04:15Z</updated>)"
+	R"(</entry>)"
+	R"(</feed>)";
+
+TEST_CASE("parse_url() converts data if specified in xml encoding attribute", "[rsspp::Parser]")
+{
+	using namespace newsboat;
+
+	constexpr auto title_utf8 = u8"タイトル"; // Japanese for "title"
+
+	auto feed_xml_utf8 = strprintf::fmt(atom_feed_with_encoding, "utf-16", title_utf8);
+
+	auto feed_xml_utf16 = utils::convert_text(feed_xml_utf8, "utf-16", "utf-8");
+	auto feed_xml = std::vector<std::uint8_t>(feed_xml_utf16.begin(), feed_xml_utf16.end());
+
+	auto& test_server = test_helpers::HttpTestServer::get_instance();
+	auto mock_registration = test_server.add_endpoint("/feed", {}, 200, {
+		{"content-type", "text/xml"},
+	}, feed_xml);
+	const auto address = test_server.get_address();
+	const auto url = strprintf::fmt("http://%s/feed", address);
+
+	rsspp::Parser parser;
+	CurlHandle easyhandle;
+	auto parsed_feed = parser.parse_url(url, easyhandle);
+
+	REQUIRE(parsed_feed.items.size() == 1);
+	REQUIRE(parsed_feed.title == title_utf8);
+}
+
+TEST_CASE("parse_url() only converts once even when encoding specified twice (xml encoding and http header)", "[rsspp::Parser]")
```

Comment:
These last 2 tests might seem a bit useless now as our implementation does not yet look at the `charset=...` part of `Content-Type` headers.
However, I expect to introduce that soon and want to use these tests to avoid regressions.